connector/servicegraphconnector/factory.go

```diff
@@ -1,13 +1,65 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-//go:generate mdatagen metadata.yaml
-
 package servicegraphconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector"
 
 import (
+	"context"
+	"time"
+
+	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/featuregate"
+
 	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector/internal/metadata"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/servicegraphprocessor"
 )
 
-var NewFactory = servicegraphprocessor.NewConnectorFactoryFunc(metadata.Type, metadata.TracesToMetricsStability)
+const (
+	// The value of "type" key in configuration.
+	typeStr = "servicegraph"
+	// The stability level of the connector.
+	connectorStability       = component.StabilityLevelDevelopment
+	virtualNodeFeatureGateID = "connector.servicegraph.virtualNode"
+)
+
+var virtualNodeFeatureGate *featuregate.Gate
+
+func init() {
+	virtualNodeFeatureGate = featuregate.GlobalRegistry().MustRegister(
+		virtualNodeFeatureGateID,
+		featuregate.StageAlpha,
+		featuregate.WithRegisterDescription("When enabled, when the edge expires, connector checks if it has peer attributes(`db.name, net.sock.peer.addr, net.peer.name, rpc.service, http.url, http.target`), and then aggregate the metrics with virtual node."),
+		featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17196"),
+	)
```

Comment:
IMO, it's a good opportunity to remove this gate and instead default to a smaller number of attributes or none

Reply:
thanks @mapno that sounds good to me. I will make these changes and request another review.